src/tracing/node_trace_writer.cc

```diff
@@ -57,9 +58,10 @@ NodeTraceWriter::~NodeTraceWriter() {
 void NodeTraceWriter::OpenNewFileForStreaming() {
   ++file_num_;
   uv_fs_t req;
-  std::ostringstream log_file;
-  log_file << "node_trace." << file_num_ << ".log";
-  fd_ = uv_fs_open(tracing_loop_, &req, log_file.str().c_str(),
+
+  char log_file[1024];
+  snprintf(log_file, 1024, log_file_pattern_, file_num_, uv_os_getpid());
```

Comment:
Since there's no validation of `log_file_pattern` things can go bad here. e.g., `./node --trace-events-enabled --trace-event-file-pattern 'tracelog.%s.log' -e 'setTimeout(() => { for (var i = 0; i < 100000; i++) { "test" + i } }, 1)'` core dumps when tried locally.

Reply:
I know. But it is a CLI flag, we document what the user should use, and it is an advanced feature. Adding validation would add a lot of complexity here.